packages/common/log/src/processors/console-processor.ts

```diff
@@ -63,17 +66,29 @@ export const DEFAULT_FORMATTER: Formatter = (config, { path, line, level, messag
     instance = chalk.magentaBright(`${prototype.constructor.name}#${id}`);
   }
 
-  return [
-    // NOTE: File path must come fist for console hyperlinks.
-    // Must not truncate for terminal output.
-    filepath,
-    column && filepath ? ''.padStart(column - filepath.length) : undefined,
-    chalk[LEVEL_COLORS[level]](column ? shortLevelName[level] : LogLevel[level]),
-    instance,
-    message,
-    context,
-    error,
-  ];
+  return config.options?.formatter?.timestampFirst
+    ? [
+        config.options?.formatter?.timestamp ? new Date().toISOString() : undefined,
+        filepath,
+        column && filepath ? ''.padStart(column - filepath.length) : undefined,
+        chalk[LEVEL_COLORS[level]](column ? shortLevelName[level] : LogLevel[level]),
+        instance,
+        message,
+        context,
+        error,
+      ]
+    : [
+        // NOTE: File path must come fist for console hyperlinks.
+        // Must not truncate for terminal output.
+        filepath,
+        column && filepath ? ''.padStart(column - filepath.length) : undefined,
+        config.options?.formatter?.timestamp ? new Date().toISOString() : undefined,
+        chalk[LEVEL_COLORS[level]](column ? shortLevelName[level] : LogLevel[level]),
+        instance,
+        message,
+        context,
+        error,
+      ];
```

Comment:
you can do something like to reduce duplication

```suggestion
const parts = [
        filepath,
        column && filepath ? ''.padStart(column - filepath.length) : undefined,
        chalk[LEVEL_COLORS[level]](column ? shortLevelName[level] : LogLevel[level]),
        instance,
        message,
        context,
        error,
      ]
   if(timestamptFirst) {
     parts.prepend(xxx);
   ]
   return parts
```